build/BUILD.zlib

```diff
@@ -0,0 +1,194 @@
+# This config closely follows the original GN build file
+# Ref: https://chromium.googlesource.com/chromium/src/third_party/zlib.git/+/refs/heads/main/BUILD.gn
+# The main difference is that we don't set ZLIB_DLL because it messes things up on Windows
+
+package(default_visibility = ["//visibility:public"])
+
+zlib_warnings = [
+    "-Wno-incompatible-pointer-types",
+    "-Wunused-variable",
+]
+
+cc_library(
+    name = "zlib_common_headers",
+    hdrs = [
+        "chromeconf.h",
+        "deflate.h",
+        "inffast.h",
+        "inffixed.h",
+        "inflate.h",
+        "inftrees.h",
+        "zconf.h",
+        "zlib.h",
+        "zutil.h",
+    ],
+    defines = ["ZLIB_IMPLEMENTATION"],
+    includes = ["."],
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "x86_defines",
+    defines = select({
+        "@platforms//os:windows": ["X86_WINDOWS"],
+        "//conditions:default": ["X86_NOT_WINDOWS"],
+    }),
+)
+
+cc_library(
+    name = "zlib_adler32_simd",
+    srcs = [
+        "adler32_simd.c",
+        "adler32_simd.h",
+    ],
+    copts = select({
+        "@platforms//cpu:x86_64": ["-mssse3"],
+        "//conditions:default": [],
+    }),
+    defines = select({
+        "@platforms//cpu:x86_64": ["ADLER32_SIMD_SSSE3"],
+        "@platforms//cpu:aarch64": ["ADLER32_SIMD_NEON"],
+    }),
+    visibility = ["//visibility:public"],
+    deps = [":zlib_common_headers"] + select({
+        "@platforms//cpu:x86_64": [":x86_defines"],
+        "//conditions:default": [],
+    }),
+)
+
+cc_library(
+    name = "zlib_arm_crc32",
+    srcs = [
+        "crc32_simd.c",
+        "crc32_simd.h",
+    ],
+    defines = ["CRC32_ARMV8_CRC32"] + select({
+        "@platforms//os:linux": ["ARMV8_OS_LINUX"],
+        "@platforms//os:macos": ["ARMV8_OS_MACOS"],
+        "@platforms//os:windows": ["ARMV8_OS_WINDOWS"],
+    }),
+    visibility = ["//visibility:public"],
+    deps = [":zlib_common_headers"],
+)
+
+cc_library(
+    name = "zlib_inflate_chunk_simd",
+    srcs = [
+        "contrib/optimizations/chunkcopy.h",
+        "contrib/optimizations/inffast_chunk.c",
+        "contrib/optimizations/inffast_chunk.h",
+        "contrib/optimizations/inflate.c",
+    ],
+    copts = zlib_warnings,
+    defines = select({
+        "@platforms//cpu:x86_64": [
+            "INFLATE_CHUNK_SIMD_SSE2",
+            "INFLATE_CHUNK_READ_64LE",
+        ],
+        "@platforms//cpu:aarch64": [
+            "INFLATE_CHUNK_SIMD_NEON",
+            "INFLATE_CHUNK_READ_64LE",
+        ],
+    }),
+    visibility = ["//visibility:public"],
+    deps = [":zlib_common_headers"],
+)
+
+cc_library(
+    name = "zlib_crc32_simd",
+    srcs = [
+        "crc32_simd.c",
+        "crc32_simd.h",
+        "crc_folding.c",
+    ],
+    copts = [
```

Comment:
This adds a requirement on ISA extensions that we didn't have before – IMHO this is reasonable since these extensions have been around for a long time but we need to document it.
If we're doing that it's reasonable to define the same set of extensions for the entire build and remove the individual flags from here (I agree that we should follow the structure of BUILD.gn for the individual targets but to make the file less complex we should not have to define ISA extension flags for each target). I will prepare a PR for this.